iroh-js/src/doc.rs

```diff
@@ -0,0 +1,1302 @@
+            .map_ok(|e| e.into())
+            .try_collect::<Vec<_>>()
+            .await?;
+        Ok(entries)
+    }
+
+    /// Get the latest entry for a key and author.
+    #[napi]
+    pub async fn get_one(&self, query: &Query) -> Result<Option<Entry>> {
+        let res = self
+            .inner
+            .get_one(query.0.clone())
+            .await
+            .map(|e| e.map(|e| e.into()))?;
+        Ok(res)
+    }
+
+    /// Share this document with peers over a ticket.
+    #[napi]
+    pub async fn share(&self, mode: ShareMode, addr_options: AddrInfoOptions) -> Result<DocTicket> {
+        let res = self
+            .inner
+            .share(mode.into(), addr_options.into())
+            .await
+            .map(|ticket| ticket.into())?;
+        Ok(res)
+    }
+
+    /// Start to sync this document with a list of peers.
+    #[napi]
+    pub async fn start_sync(&self, peers: Vec<NodeAddr>) -> Result<()> {
+        let peers = peers
+            .into_iter()
+            .map(|p| p.try_into())
+            .collect::<anyhow::Result<Vec<_>>>()?;
+
+        self.inner.start_sync(peers).await?;
+        Ok(())
+    }
+
+    /// Stop the live sync for this document.
+    #[napi]
+    pub async fn leave(&self) -> Result<()> {
+        self.inner.leave().await?;
+        Ok(())
+    }
+
+    /// Subscribe to events for this document.
+    #[napi]
+    pub async fn subscribe(&self, cb: ThreadsafeFunction<LiveEvent, ()>) -> Result<()> {
+        let client = self.inner.clone();
+        tokio::task::spawn(async move {
+            let mut sub = client.subscribe().await.unwrap();
+            while let Some(event) = sub.next().await {
+                let message: Result<LiveEvent> = event.map(Into::into).map_err(Into::into);
+                if let Err(err) = cb.call_async(message).await {
+                    warn!("cb error: {:?}", err);
+                }
+            }
+        });
+
+        Ok(())
+    }
+
+    /// Get status info for this document
+    #[napi]
+    pub async fn status(&self) -> Result<OpenState> {
+        let res = self.inner.status().await.map(|o| o.into())?;
+        Ok(res)
+    }
+
+    /// Set the download policy for this document
+    #[napi]
+    pub async fn set_download_policy(&self, policy: &DownloadPolicy) -> Result<()> {
+        self.inner.set_download_policy(policy.into()).await?;
+        Ok(())
+    }
+
+    /// Get the download policy for this document
+    #[napi]
+    pub async fn get_download_policy(&self) -> Result<DownloadPolicy> {
+        let res = self
+            .inner
+            .get_download_policy()
+            .await
+            .map(|policy| policy.into())?;
+        Ok(res)
+    }
+
+    /// Get sync peers for this document
+    #[napi]
+    pub async fn get_sync_peers(&self) -> Result<Option<Vec<Vec<u8>>>> {
+        let list = self.inner.get_sync_peers().await?;
+        let list = list.map(|l| l.into_iter().map(|p| p.to_vec()).collect());
+        Ok(list)
+    }
```

Comment:
is the value that comes back from this the bytes of the public key? Do I need to base32 these myself up in JS land?